evaluator.plandefinition/src/main/java/org/opencds/cqf/cql/evaluator/plandefinition/r4/PlanDefinitionProcessor.java

```diff
@@ -0,0 +1,725 @@
+    if (dataRequirements == null || dataRequirements.isEmpty()) {
+      return null;
+    }
+    Parameters params = new Parameters();
+    for (DataRequirement req : dataRequirements) {
+      String name = req.getId();
+      String fhirType = req.getType();
+      Iterator<IBaseResource> searchResult = fhirDal.search(fhirType).iterator();
+      if (searchResult != null && searchResult.hasNext()) {
+        int index = 0;
+        Boolean found = true;
+        while (searchResult.hasNext()) {
+          Resource resource = (Resource)searchResult.next();
+          ParametersParameterComponent parameter = new ParametersParameterComponent().setName("%" + String.format("%s", name));
+          if (req.hasCodeFilter()) {
+            for (DataRequirement.DataRequirementCodeFilterComponent filter : req.getCodeFilter()) {
+              Parameters codeFilterParam = new Parameters();
+              codeFilterParam.addParameter().setName("%resource").setResource(resource);
+              if (filter != null && filter.hasPath() && filter.hasValueSet()) {
+                Iterable<IBaseResource> valueset = fhirDal.searchByUrl("ValueSet", filter.getValueSet());
+                System.out.println(filter.getValueSet());
+                if (valueset != null && valueset.iterator().hasNext()) {
+                  codeFilterParam.addParameter().setName("%valueset").setResource((Resource)valueset.iterator().next());
+                  String codeFilterExpression = "%" + String.format("resource.%s.where(code.memberOf(\'%s\'))", filter.getPath(), "%" + "valueset");
+                  IBaseParameters codeFilterResult = expressionEvaluator.evaluate(codeFilterExpression, codeFilterParam);
+                  IBaseDatatype tempResult = operationParametersParser.getValueChild(((Parameters) codeFilterResult), "return");
+                  if (tempResult != null) {
+                    if (tempResult instanceof BooleanType) {
+                      found = ((BooleanType)tempResult).booleanValue();
+                    }
+                  }
+                }
+                logger.debug(String.format("Could not find ValueSet with url %s on the local server.", filter.getValueSet()));
+              }
+            }
+          }
+          if (!searchResult.hasNext() && index == 0) {
+            parameter.addExtension("http://hl7.org/fhir/uv/cpg/StructureDefinition/cpg-parameterDefinition", new ParameterDefinition().setMax("*").setName("%" + name));
+            if (found) {
+              parameter.setResource(resource);
+            }
+          } else {
+            if (!found) {
+              index++;
+              continue;
+            }
+            parameter.setResource(resource);
+          }
+          params.addParameter(parameter);
+          index++;
+        }
+      } else {
+        ParametersParameterComponent parameter = new ParametersParameterComponent().setName("%" + String.format("%s", name));
```

Comment:
For inputs coming from data requirements you can just always add the extension marking it as a List.